org/postgresql/jdbc2/TypeInfoCache.java

```diff
@@ -150,9 +153,14 @@ public synchronized void addCoreType(String pgTypeName, Integer oid, Integer sql
         Character delim = ',';
         _arrayOidToDelimiter.put(oid, delim);
 
-        String pgArrayTypeName = "_" + pgTypeName;
```

Comment:
_ prefix is not fully correct way for referencing types. Look at this example:

create type _custom as (f float);
create type custom as (i int);
select * from pg_type where typname like '%custom'

number of _ prefixes depends on the creation order and _ does not imply array.

There is special alias lookup where are builtin types should be registered to support such scenario: https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java#L104

Basically, [] at the end of type is the only fully correct way to represent arrays.